code/common/os.q

```diff
@@ -2,7 +2,7 @@
 \d .os
 NT:.z.o in`w32`w64
 Fex:{not 0h~type key hsym$[10=type x;`$x;x]}
-pth:{if[10h<>type x;x:string x]; if[NT;x:@[x;where"/"=x;:;"\\"]];$[":"=first x;1_x;x]}
+pth:{if[10h<>type x;x:string x]; if[NT;x:@[x;where"/"=x;:;"\\"]];(":"=first x)_x}
 ext:{`$(x?".")_(x:string x;x)[i:10h=type x]}
```

Comment:
This function is a bit odd, it always strings `x`, whether or not `i` is true (and `i` is never used), although it seems to work regardless. It should probably match the other functions here:
```
{if[10h<>type x;x:string x];`$(x?".")_x}
```
Or simpler as per @jonathonmcmurray's suggestion:
```
{`$(x?".")_x:(string x;x)10h=type x}
```
If you don't want to update the logic at least remove `i` as it is unused.

Reply:
I think that's what the function is supposed to achieve. Returning the extension means that when dealing with archived files it should return `.tar.gz`. I'll modify it as you've recommended.